utils/windowfunction/wf_nth_value.cpp

```diff
@@ -84,31 +84,19 @@ boost::shared_ptr<WindowFunctionType> WF_nth_value<T>::makeFunction(int id, cons
         }
 
         case CalpontSystemCatalog::DECIMAL:
+        case CalpontSystemCatalog::UDECIMAL:
         {
-            if (wc->functionParms()[0]->resultType().colWidth < 16)
+            if (wc->functionParms()[0]->resultType().colWidth < datatypes::MAXDECIMALWIDTH)
             {
                 func.reset(new WF_nth_value<int64_t>(id, name));
             }
-            else
+            else if (wc->functionParms()[0]->resultType().colWidth == datatypes::MAXDECIMALWIDTH)
             {
                 func.reset(new WF_nth_value<int128_t>(id, name));
             }
             break;
         }
         
-        case CalpontSystemCatalog::UDECIMAL:
-        {
-            if (wc->functionParms()[0]->resultType().colWidth < 16)
-            {
-                func.reset(new WF_nth_value<uint64_t>(id, name));
-            }
-            else
-            {
-                func.reset(new WF_nth_value<uint128_t>(id, name));
-            }
-            break;
-        }
-
```

Comment:
I am not sure if removing the separate case for UDECIMAL is correct now. As earlier we had `func.reset(new WF_nth_value<uint64_t>(id, name));` for UDECIMAL, now we have `func.reset(new WF_nth_value<int64_t>(id, name));`. Similarly for the wide decimal case.

@dhall-MariaDB can you confirm this change is correct?

Reply:
Why exactly are you not sure?
My points are: 
- IDB_Decimal uses int64 to store the value and it is enough to store decimal/udecimal(18)
- All WFs changed(except avg) can't produce the value bigger than column can have. avg doesn't use the signedness
- If MCS runs WF on expression/aggregate there are overflow checks in the relevant code.

Reply:
I didn't have the full context on the changes here, thanks for this clarification.